components/script/microtask.rs

```diff
@@ -28,6 +32,13 @@ pub struct MicrotaskQueue {
 #[derive(JSTraceable, HeapSizeOf)]
 pub enum Microtask {
     Promise(EnqueuedPromiseCallback),
+    AwaitStableState(Box<Runnable + Send>)
```

Comment:
Instead of using Runnables for this and await_stable_state, let's have `MedieElement(MediaElementMicrotask)`, `ImageElement(ImageElementMicrotask)`, where the inner types are enums. We can define a `MicrotaskRunnable` trait that the enums implement with a `handler(&self)` method.